read_test.go

```diff
@@ -595,6 +595,141 @@ func TestReadPixelData_SkipPixelData(t *testing.T) {
+		mustNewElement(tag.SOPInstanceUID, []string{"1.3.6.1.4.1.35190.4.1.20210608.607733549593"}),
+	}
+	data, err := writeElements(elements)
+	if err != nil {
+		return nil, err
+	}
+
+	// Construct valid DICOM header preamble.
+	magicWord := []byte("DICM")
+	preamble := make([]byte, 128)
+	preamble = append(preamble, magicWord...)
+	headerBytes := append(preamble, data...)
+	headerData.HeaderBytes = bytes.NewBuffer(headerBytes)
+	headerData.Elements = elements[0 : len(elements)-1]
+	return headerData, nil
+}
+
+// Returns a fake DICOM group 2 header with a FileMetaInformationGroupLength tag (0x0002,0x0000).
+func headerWithFileMetaInformationGroupLength() (*headerData, error) {
+	headerData := new(headerData)
+
+	sopInstanceUidElement := mustNewElement(tag.SOPInstanceUID, []string{"1.3.6.1.4.1.35190.4.1.20210608.607733549593"})
+	elements := []*Element{
+		mustNewElement(tag.FileMetaInformationVersion, []byte{0x00, 0x01}),
+		mustNewElement(tag.MediaStorageSOPClassUID, []string{"1.2.276.0.7230010.3.1.0.1"}),
+		mustNewElement(tag.MediaStorageSOPInstanceUID, []string{"1.3.6.1.4.1.35190.4.1.20210608.607733549593"}),
+		mustNewElement(tag.TransferSyntaxUID, []string{"=RLELossless"}),
+		mustNewElement(tag.ImplementationClassUID, []string{"1.2.276.0.7230010.3.0.3.6.7"}),
+		mustNewElement(tag.ImplementationVersionName, []string{"OFFIS_DCMTK_367"}),
+	}
+	dataHeader, err := writeElements(elements)
+	if err != nil {
+		return nil, err
+	}
+	fileMetaInfoElement := mustNewElement(tag.FileMetaInformationGroupLength, []int{len(dataHeader)})
+	dataFileMetaInfo, err := writeElements([]*Element{fileMetaInfoElement})
+	if err != nil {
+		return nil, err
+	}
+	dataSopInstanceUid, err := writeElements([]*Element{sopInstanceUidElement})
+	if err != nil {
+		return nil, err
+	}
+	data := append(dataFileMetaInfo, dataHeader...)
+	data = append(data, dataSopInstanceUid...)
+
+	// Construct valid DICOM header preamble.
+	magicWord := []byte("DICM")
+	preamble := make([]byte, 128)
+	preamble = append(preamble, magicWord...)
+	headerBytes := append(preamble, data...)
+	headerData.HeaderBytes = bytes.NewBuffer(headerBytes)
+	headerData.Elements = append([]*Element{fileMetaInfoElement}, elements...)
+	return headerData, nil
+}
+
+func TestReadHeader_TryAllowErrorMetaElementGroupLength(t *testing.T) {
+	opts := parseOptSet{allowErrorMetaElementGroupLength: true}
+
+	t.Run("NoFileMetaInformationGroupLength", func(t *testing.T) {
+		dcmheaderNoInfoGrpLen, err := headerWithNoFileMetaInformationGroupLength()
+		if err != nil {
+			t.Fatalf("unsuccesful generation of fake header data")
+		} else {
+			r := &reader{
+				rawReader: dicomio.NewReader(bufio.NewReader(dcmheaderNoInfoGrpLen.HeaderBytes), binary.LittleEndian, int64(dcmheaderNoInfoGrpLen.HeaderBytes.Len())),
+				opts:      opts,
+			}
+			r.rawReader.SetTransferSyntax(binary.LittleEndian, true)
+			parsedElements, err := r.readHeader()
```

Comment:
nit: maybe we can call this `wantElements` instead of parsedElements (in the test below too).